### DISCUSSION TITLE
how I can compile zfs module directly with kernel?

### DISCUSSION THREAD

**Discussion by osevan** [openzfs/zfs#13455] [Q&A]
I know its possible. 

Any scripts existing to move sources of zfs and make instructions to Linux build three? 

Thanks and

Best regards

**Comment by Dragon-debian** [comment]
This might help.
https://openzfs.github.io/openzfs-docs/Developer%20Resources/Building%20ZFS.html

**Comment by IvanVolosyuk** [comment]
Don't see any reference to this in the doc. Here is the Gentoo way:
https://wiki.gentoo.org/wiki/ZFS#Installing_into_the_kernel_directory_.28for_static_installs.29
Basically boils down to:
```
..
./configure --enable-linux-builtin
..
./copy-builtin /usr/src/linux
..
```
Add ```CONFIG_SPL=y``` and ```CONFIG_ZFS=y``` in the kernel's .config
```$ make oldconfig```
in the kernel tree should help with that.
